manager/utils.py

```diff
@@ -113,141 +118,108 @@ def import_emails(self):
 
         csv_stream = StringIO(csv_string)
 
-        csv_reader = csv.reader(csv_stream)
+        csv_reader = csv.DictReader(csv_stream, fieldnames=columns)
 
-        email_address_index = columns.index('email')
-        try:
-            first_name_index = columns.index('first_name')
-        except ValueError:
-            first_name_index = None
-        try:
-            last_name_index = columns.index('last_name')
-        except ValueError:
-            last_name_index = None
-        try:
-            preferred_name_index = columns.index('preferred_name')
-        except ValueError:
-            preferred_name_index = None
+        # For duplication tracking
+        processed_emails = []
+
+        recipient_creates = []
+        recipient_updates = []
+
+        attribute_creates = []
+        attribute_updates = []
+
+        created_objs = []
 
         for idx, row in enumerate(csv_reader):
+            if self.skip_first_row and idx == 0: continue
+
             row_num = idx + 1
 
-            if idx == 0 and self.skip_first_row == True:
-                continue
+            try:
+                email_address = row['email'].lower()
```

Comment:
I'm not sure if csv's DictReader will automatically trim strings in cell data--it may be worth checking and calling `.strip()` on `email_address`, `first_name`, `last_name` and `preferred_name` if it doesn't to ensure empty string checks (e.g. on line 149) don't fail when a cell only contains whitespace.